### ISSUE TITLE
Migration guide

### ISSUE THREAD

**Issue by kyazdani42** [kyazdani42/nvim-tree.lua#674]
After migrating to the new setup, these are the variables that have been migrated to the setup configuration table:

- `nvim_tree_disable_netrw` -> `disable_netrw`
- `nvim_tree_hijack_netrw`  -> `hijack_netrw`
- `nvim_tree_auto_open`   -> `open_on_setup`
- `nvim_tree_auto_close`   -> `auto_close`
- `nvim_tree_tab_open`    -> `open_on_tab`
- `nvim_tree_update_cwd` -> `update_cwd`
- `nvim_tree_hijack_cursor`   -> `hijack_cursor`
- `nvim_tree_system_open_command`  -> `system_open.cmd`
- `nvim_tree_system_open_command_args` -> `system_open.args`
- `nvim_tree_follow`   -> `update_focused_file.enable`
- `nvim_tree_follow_update_path` -> `update_focused_file.update_cwd`
- `nvim_tree_lsp_diagnostics` -> `lsp_diagnostics`
- `nvim_tree_auto_resize` -> `view.auto_resize`
- `nvim_tree_bindings` -> `view.mappings.list`
- `nvim_tree_hide_dotfiles` -> `filters.dotfiles`
- `nvim_tree_ignore` -> `filters.custom`
- `nvim_tree_gitignore` -> `git.ignore`

`nvim_tree_disable_keybindings` and `nvim_tree_disable_default_keybindings` have been replaced by using `view.mappings.custom_only`.
Setting `view.mappings.custom_only` to `true` and `view.mappings.list` to `{}` would do the same as `nvim_tree_disable_keybindings` and `view.mappings.custom_only` to `true` with your list of keybindings would do the same as `nvim_tree_disable_default_keybindings`.

NOTE: you need to set other `vim.g` `let g:` nvim_tree variables _BEFORE_ calling the `setup` if you want everything to work as expected :)

**Comment by wookayin** [comment]
Please pin this or add a link README!

**Comment by JarnoB** [comment]
Thanks, this is helpful! 
But isn't `nvim_tree_tab_open -> tab_open` supposed to be `nvim_tree_tab_open -> open_on_tab`?

**Comment by kyazdani42** [comment]
good catch !

**Comment by ismikhail** [comment]
The `view` config does not seem to be working for me. 
I installed the plugin with Packer like this:
```
  use {
    'kyazdani42/nvim-tree.lua',
    requires = 'kyazdani42/nvim-web-devicons'
  }
```
Then use this setup
```
require'nvim-tree'.setup {
  auto_close = true,
  view = {
    side = 'right',
    width = 70
  }
}
```
The `auto_close` setting works but the `view` does not have any effect.

**Comment by nshen** [comment]
> The `view` config does not seem to be working for me. I installed the plugin with Packer like this:
> 
> ```
>   use {
>     'kyazdani42/nvim-tree.lua',
>     requires = 'kyazdani42/nvim-web-devicons'
>   }
> ```
> 
> Then use this setup
> 
> ```
> require'nvim-tree'.setup {
>   auto_close = true,
>   view = {
>     side = 'right',
>     width = 70
>   }
> }
> ```
> 
> The `auto_close` setting works but the `view` does not have any effect.

same here. I can't set width,  in either way.

**Comment by sgalizia** [comment]
Just to chime in here, I am also unable to get the width set using the above two comments method in the setup method.

**Comment by b0o** [comment]
It would be nice if you added a link to this issue inside the deprecation warning messages. You could use GitHub's [`git.io`](https://git.io/) URL shortener, e.g. [`https://git.io/JPhyt`](https://git.io/JPhyt).

**Comment by kyazdani42** [comment]
@ismikhail @nshen @sgalizia i cannot reproduce, could you open a new issue if this is still happening to you please ?

**Comment by lockejan** [comment]
> > The `view` config does not seem to be working for me. I installed the plugin with Packer like this:
> > ```
> >   use {
> >     'kyazdani42/nvim-tree.lua',
> >     requires = 'kyazdani42/nvim-web-devicons'
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then use this setup
> > ```
> > require'nvim-tree'.setup {
> >   auto_close = true,
> >   view = {
> >     side = 'right',
> >     width = 70
> >   }
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The `auto_close` setting works but the `view` does not have any effect.
> 
> same here. I can't set width, in either way.

Have you tried disabling other plugins to make sure there is nothing interfering?
Because your code works as expected on my side.

**Comment by sgalizia** [comment]
> @ismikhail @nshen @sgalizia i cannot reproduce, could you open a new issue if this is still happening to you please ?

I finally figured out my issue today. The tree was using the custom width I set only after toggling it on and off. The `auto_resize` setting in the view options defaulting to `false` is what tripped me up. After setting that to true, the tree behaved how I expected it to. I would possibly suggest a default of true for that setting? Otherwise, opening a buffer from the tree causes the tree buffer to take up half the screen (vsplit) which is odd behavior for it in my opinion.

**Comment by kyazdani42** [comment]
closing this, it's a bit old now.

**Comment by wookayin** [comment]
Just for those who might find this issue via README: Since #941 (230a61d), neovim 0.5.0 will not work due to the use of new option. If somehow you'd need to use neovim 0.5.x, d1ff893 would be a version that will work for neovim 0.5.0.

**Comment by Meigumikato** [comment]
i set  filters.dotfiles = true do not  effect work, but when i set nvim_tree_hide_dotfiles = 1, it was effect

>`require'nvim-tree'.setup {
  diagnostics = {
    enable = false,
    icons = {
      hint = "",
      info = "",
      warning = "",
      error = ""
    }
  },
  filters = {dotfiles = true},
}
vim.g.nvim_tree_hide_dotfiles = 1
`

**Comment by alex-courtis** [comment]
> i set filters.dotfiles = true do not effect work, but when i set nvim_tree_hide_dotfiles = 1, it was effect

I have tested `filters = {dotfiles = true},` using current master 5015e72 and it is functioning correctly. Which version are you running @Meigumikato ? 

If you are behind, please upgrade and test again. Raise a [new issue](https://github.com/kyazdani42/nvim-tree.lua/issues) if the problem persists.

**Comment by ttys3** [comment]
`nvim_tree_indent_markers` =>  `renderer.indent_markers.enable`

**Comment by b0o** [comment]
> It would be nice if you added a link to this issue inside the deprecation warning messages. If you think the URL is too long, you could use GitHub's [`git.io`](https://git.io/) URL shortener, e.g. [`git.io/JPhyt`](https://git.io/JPhyt). Alternatively, you could add this information to `nvim-tree-lua.txt`.

Just wanted to mention that GitHub [is retiring](https://github.blog/changelog/2022-04-25-git-io-deprecation/) git.io, and it will be shut down in 3 days, in case this link is still in use.

**Comment by alex-courtis** [comment]
> > It would be nice if you added a link to this issue inside the deprecation warning messages. If you think the URL is too long, you could use GitHub's [`git.io`](https://git.io/) URL shortener, e.g. [`git.io/JPhyt`](https://git.io/JPhyt). Alternatively, you could add this information to `nvim-tree-lua.txt`.
> 
> Just wanted to mention that GitHub [is retiring](https://github.blog/changelog/2022-04-25-git-io-deprecation/) git.io, and it will be shut down in 3 days, in case this link is still in use.

#1209

**Comment by sarmong** [comment]
Since `auto_close` has been completely removed, better to remove it from this list as well or tell that it can be done with custom autocommand

**Comment by kyazdani42** [comment]
see https://github.com/kyazdani42/nvim-tree.lua/discussions/1115

**Comment by AlvaroPata** [comment]
Hi! Is there a reason why I still get this warning even with the default configuration?

**Comment by alex-courtis** [comment]
> Hi! Is there a reason why I still get this warning even with the default configuration?

Are you receiving

1. "Following options were moved to setup..." warning
2. "unknown/invalid option".

For 2, please see #1334

1 indicates that you have global `g:` variabled defined somewhere.

**Comment by AlvaroPata** [comment]
Thanks! I found some g. settings i had forgotten about!

**Comment by dongwangdw** [comment]
I can not understand here. I use the latest version and I got the warning each time I open nvim. Should I do some manually work here?

**Comment by kyazdani42** [comment]
You might have some `g:nvim_tree...` options left. What does the warning print ??

**Comment by dongwangdw** [comment]
@kyazdani42 
Thanks for your reply. I have resolved this. I need to modify my config accordingly.
